vendor/engines/bulk_email/app/controllers/bulk_email/bulk_email_controller.rb

```diff
@@ -60,7 +60,20 @@ def bulk_email_cancel_path
     end
 
     def cancel_params
-      @cancel_params ||= params.slice(:start_date, :end_date, :bulk_email, :products)
+      @cancel_params ||= params.slice(:start_date, :end_date, :bulk_email, :products, :return_path)
+    end
+
+    def delivery_success_path
+      return_path_from_params || facility_bulk_email_path
+    end
+
+    def return_path_from_params
+      return_path = params[:return_path].presence
+      return_path if return_path &&
```

Comment:
I think a comment about why this is doing these things would be a good idea so future us knows that this is about sanitization.